src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java

```diff
@@ -199,13 +290,324 @@ public String getContentTypeTemplatePath(String siteId, String contentTypeId) th
+     * @param config {@link ContentTypeConfigTO} object to check
+     * @param contentTypes list of {@link ContentTypeConfigTO} to add
+     */
+    protected void addContentTypes(Set<String> userRoles, ContentTypeConfigTO config,
+                                   List<ContentTypeConfigTO> contentTypes) {
+        boolean isAllowed = this.isUserAllowed(userRoles, config);
+        if (isAllowed) {
+            contentTypes.add(config);
+        }
+    }
+
+    /**
+     * Check if user is allowed to access given content type
+     * @param userRoles list of user roles
+     * @param item {@link ContentTypeConfigTO} object to check
+     * @return true if allowed, false otherwise
+     */
+    protected boolean isUserAllowed(Set<String> userRoles, ContentTypeConfigTO item) {
+        if (item == null) {
+            logger.debug("No content type config provided for null item to limit user access, " +
+                    "defaulting to permit the user");
+            return true;
+        }
+
+        Set<String> allowedRoles = item.getAllowedRoles();
+        logger.trace("Item '{}' allows roles '{}', checking against user roles '{}'",
+                item.getName(), allowedRoles, userRoles);
+
+        if (allowedRoles == null || allowedRoles.size() == 0) {
+            logger.trace("User with roles '{}' is allowed access to '{}'", userRoles, item.getName());
+            return true;
+        }
+
+        boolean notAllowed = Collections.disjoint(userRoles, allowedRoles);
+        if (notAllowed) {
+            logger.debug("Item '{}' is not allowed for user with roles '{}'",
+                    item.getName(), userRoles);
+            return false;
+        }
+
+        logger.trace("User with roles '{}' is allowed access to '{}'", userRoles, item.getName());
+        return true;
+    }
+
+    /**
+     * Get location where content types configuration files are stored for a site
+     * @return content type path in String
+     */
+    protected String getConfigPath() {
+        return studioConfiguration.getProperty(CONFIGURATION_SITE_CONTENT_TYPES_CONFIG_PATH);
+    }
+
+    /**
+     * Get file name where configuration of content type is stored
+     * @return file name in String
+     */
+    protected String getConfigFileName() {
+        return studioConfiguration.getProperty(CONFIGURATION_SITE_CONTENT_TYPES_CONFIG_FILE_NAME);
+    }
+
+    /**
+     * Get location where content types are stored for a site
+     * @return path in string
+     */
+    public String getSiteContentTypesConfigPath() {
+        return studioConfiguration.getProperty(CONFIGURATION_SITE_CONTENT_TYPES_CONFIG_BASE_PATH);
+    }
+
+    /**
+     * get paths
+     *
+     * @param root element
+     * @param path path to get
+     * @return list of paths
+     */
+    private List<String> getPaths(Element root, String path) {
+        List<String> paths;
+        List<Node> nodes = root.selectNodes(path);
+        if (nodes != null && nodes.size() > 0) {
```

Comment:
Please use guard statements
Consider using `CollectionUtils.isEmpty` in these scenarios